drivers/src/qemuexit.rs

```diff
@@ -6,18 +6,26 @@ use crate::Driver;
 const RISCV64_BASE_ADDRESS: usize = 0x100000;
 
 pub struct QemuExit {
+    #[cfg(target_arch = "riscv64")]
     address: usize,
     #[cfg(target_arch = "riscv64")]
     inner: qemu_exit::RISCV64,
+    #[cfg(target_arch = "aarch64")]
+    inner: qemu_exit::AArch64,
 }
 
 impl QemuExit {
     pub fn new() -> Self {
         #[cfg(target_arch = "riscv64")]
-        Self {
+        return Self {
             address: RISCV64_BASE_ADDRESS,
             inner: qemu_exit::RISCV64::new(RISCV64_BASE_ADDRESS as u64),
-        }
+        };
+
+        #[cfg(target_arch = "aarch64")]
+        return Self {
+            inner: qemu_exit::AArch64::new(),
+        };
```

Comment:
You don't have to use `return`s for this
```suggestion
        #[cfg(target_arch = "aarch64")]
        Self {
            inner: qemu_exit::AArch64::new(),
        }
```
(and similarly for the `riscv64` version)

should suffice. You can simply keep both expression blocks hidden behind `#[cfg]` directives unless I'm missing something